R/dataframe-frame.R

```diff
@@ -145,25 +185,132 @@ dataframe__get_columns <- function() {
     })
 }
 
+#' Group a DataFrame
+#' @inheritParams LazyFrame_group_by
+#' @inherit LazyFrame_group_by description params
+#' @details Within each group, the order of the rows is always preserved,
+#' regardless of the `maintain_order` argument.
+#' @return [GroupBy][GroupBy_class] (a DataFrame with special groupby methods like `$agg()`)
+#' @seealso
+#' - [`<DataFrame>$partition_by()`][DataFrame_partition_by]
+#' @examples
+#' df <- pl$DataFrame(
+#'   a = c("a", "b", "a", "b", "c"),
+#'   b = c(1, 2, 1, 3, 3),
+#'   c = c(5, 4, 3, 2, 1)
+#' )
+#'
+#' df$group_by("a")$agg(pl$col("b")$sum())
+#'
+#' # Set `maintain_order = TRUE` to ensure the order of the groups is consistent with the input.
+#' df$group_by("a", maintain_order = TRUE)$agg(pl$col("c"))
+#'
+#' # Group by multiple columns by passing a list of column names.
+#' df$group_by(c("a", "b"))$agg(pl$max("c"))
+#'
+#' # Or pass some arguments to group by multiple columns in the same way.
+#' # Expressions are also accepted.
+#' df$group_by("a", pl$col("b") %/% 2)$agg(
+#'   pl$col("c")$mean()
+#' )
+#'
+#' # The columns will be renamed to the argument names.
+#' df$group_by(d = "a", e = pl$col("b") %/% 2)$agg(
+#'   pl$col("c")$mean()
+#' )
 dataframe__group_by <- function(..., maintain_order = FALSE) {
   wrap_to_group_by(self, list2(...), maintain_order)
 }
 
+#' Select and modify columns of a DataFrame
+#' @description Similar to `dplyr::mutate()`. However, it discards unmentioned
+#' columns (like `.()` in `data.table`).
+#'
+#' @param ... Columns to keep. Those can be expressions (e.g `pl$col("a")`),
+#' column names  (e.g `"a"`), or list containing expressions or column names
+#' (e.g `list(pl$col("a"))`).
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' pl$DataFrame(iris)$select(
+#'   pl$col("Sepal.Length")$abs()$alias("abs_SL"),
+#'   (pl$col("Sepal.Length") + 2)$alias("add_2_SL")
+#' )
```

Comment:
I think these methods need to be inherited from lazyframe's.

Reply:
I added inherited docs for select and with_columns, but other methods in lazyframe don't have their docs yet so I'd like to add those and deduplicate the dataframe/lazyframe in a future PR